hazelcast-hibernate52/src/main/java/com/hazelcast/hibernate/local/CleanupService.java

```diff
@@ -31,19 +33,23 @@
     private final Duration fixedDelay;
     private final String name;
     private final ScheduledExecutorService executor;
+    private final List<Runnable> cacheDestroy;
```

Comment:
Wouldn't it be easier to actually store `LocalRegionCache` instances and then iterate over them and call `destroy()` on each?

Reply:
Yes, it would be simpler. Also, it would increase the coupling for future changes, the idea was to just use the needed parts and hide the rest inside the CleanupService. 
We can do it that other way too.

Reply:
I don't think that that would result in extra coupling. Keep in mind that the `registerCache()` method accepts `LocalRegionCache` objects and the rest is encapsulated within the `CleanupService` class and doesn't really leak out in any way.